### PR TITLE
Add CONTRIBUTING.md, GOVERNANCE.md, MAINTAINERS.md Docs

### PR DESCRIPTION
**Pull request by ihcsim**
## Change Overview

This PR adds the 1st iteration of the following docs to the repository:

* CONTRIBUTING.md
* GOVERNANCE.md
* MAINTAINERS.md
* [CODEOWNER](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)

Instructions on how to create and test a local build will be added to BUILD.md in a separate PR.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [x] :world_map: Documentation
- [ ] :robot: Test

## Issues

Fix https://github.com/kanisterio/kanister/issues/1119
Fix https://github.com/kanisterio/kanister/issues/1234

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+Short one line title
+
+An explanation of the problem, providing context, and why the change is being
+made.
+```
+
+### Submitting Pull Requests
+
+When submitting a pull request, it's important that you clearly communicate your
+intent, by clearly:
```
Comment (pavannd1) [inline CONTRIBUTING.md:100]: [nit] we can probably remove one `clearly` here

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+In general, smaller pull requests are easier to review and merge than bigger
+ones. It's always a good idea to collaborate with the maintainers to determine
+how best to break up a big pull request.
+
+Once the maintainers approve your PR, they will label it as `kueue`. The
+`mergify` bot will then squash the commits in your PR, and add it to the merge
+queue. The bot will auto-merge your work when it's ready.
+
+Congratulation! Your pull request has been successfully merged! 👏
```
Comment (pavannd1) [inline CONTRIBUTING.md:119]:
```suggestion
Congratulations! Your pull request has been successfully merged! 👏
```
[nit] again

File: MAINTAINERS.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+* Pavan Navarathna @pavannd1
+* Prasad Ghangal @PrasadG193
+* Vivek Singh @viveksinghggits
+* Ivan Sim @ihcsim
+
+## Maintainer Responsibilities
+
+Maintainers are active and visible members of the community. They have the a lot
+of experience with the project and are expected to have the knowledge and
+insight to lead its growth and improvement.
```
Comment (pavannd1) [inline MAINTAINERS.md:21]:
```suggestion
Maintainers are active and visible members of the community. They have a lot
of experience with the project and are expected to have the knowledge and
insight to lead its growth and improvement.
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1 @@
+* @kanisterio/maintainers
```
[the remark this answers is not in the document]
Comment (ihcsim) [reply .github/CODEOWNERS:1]: Yes, maintainers will always be added as reviewers, but their reviews and approvals are not strictly required for PRs to be merged.

[the remark this answers is not in the document]
Comment (viveksinghggits) [reply .github/CODEOWNERS:1]: ok, let's see if Pavan has some inputs.

**Comment by viveksinghggits** [comment]
LGTM

File: .github/CODEOWNERS (hunk 4 above)
[the remark this answers is not in the document]
Comment (pavannd1) [reply .github/CODEOWNERS:1]: I agree with having maintainers keep track of things getting merged 👍🏼

**Review by pavannd1 — APPROVED** [review]
LGTM 👍🏼

File: .github/CODEOWNERS (hunk 4 above)
[the remark this answers is not in the document]
Comment (viveksinghggits) [reply .github/CODEOWNERS:1]: right, I understand but I thought people are going to think someone else is going to review the change. But looks good, let's merge this :ship: .

**Review by viveksinghggits — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+When submitting a pull request, it's important that you communicate your intent,
+by clearly:
+
+1. describing the problem you are trying to solve with links to the relevant
+GitHub issues
+1. describing your solution with links to any design documentation and
+discussion
+1. defining how you test and validate your solution
+1. updating the relevant documentation and examples where appropriate
```
Comment (viveksinghggits) [inline CONTRIBUTING.md:107]: should it be numbered list or all 1s are fine.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+Finalized commit messages should look similar to the following format:
+
+```text
+Short one line title
+
+An explanation of the problem, providing context, and why the change is being
+made.
+```
+
+### Submitting Pull Requests
```
Comment (viveksinghggits) [inline CONTRIBUTING.md:97]: Hi @ihcsim 
is it a good idea to ask people to confirm with maintainers if the pull request that they are trying to raise would be a good addition. In some cases there are chances that the PR is raised but the maintainers don't think that the PR should get in.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+* adhering to guidelines found in the [Effective Go](https://go.dev/doc/effective_go) document
+* following the common [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+
+The [golangci-lint](https://golangci-lint.run/) tool is used to enforce many
+styling and safety rules.
+
+### Creating A Local Build
+
+See the [BUILD.md](BUILD.md) document for instructions on how to build, test and
```
Comment (viveksinghggits) [inline CONTRIBUTING.md:77]: are we yet to push the build.md?

File: CONTRIBUTING.md (hunk 5 above)
Comment (pavannd1) [reply CONTRIBUTING.md:107]: I think this renders correctly once merged. See preview!

File: CONTRIBUTING.md (hunk 6 above)
Comment (ihcsim) [reply CONTRIBUTING.md:97]: Absolutely. Ideally, we would have prior discussion in GitHub issues/discussions, before pull requests are submitted. That's why in this paragraph I directed contributors to look up GitHub issues first.

File: CONTRIBUTING.md (hunk 7 above)
Comment (ihcsim) [reply CONTRIBUTING.md:77]: It will be in a separate PR.

File: CONTRIBUTING.md (hunk 6 above)
Comment (pavannd1) [reply CONTRIBUTING.md:97]: I think it is okay to determine this on the PR itself. At least that's what I have seen on some other OSS repos

**Pull request merged**